src/OrchardCore/OrchardCore.Users.Core/Services/UserService.cs

```diff
@@ -370,4 +370,6 @@ public async Task<IUser> RegisterAsync(RegisterUserForm model, Action<string, st
 
         return user;
     }
+
+    public Task<IEnumerable<IUser>> GetUsersAsync() => Task.FromResult<IEnumerable<IUser>>(_userManager.Users);
```

Comment:
What if there is lots of records? Probable better to have PageAsync where you can request pages of users. Anyway, I don't see it being using in OC. Why are you adding it?

Reply:
> What if there is lots of records? Probable better to have PageAsync where you can request pages of users.

Could be, or we could add two params to support pagination, also we might return queryable of users

>  Why are you adding it?

This is a valuable method; it has been requested a few times

Reply:
I wouldn't add it to the framework since it's not used by the framework and I think it's a bad idea to allow someone to load all these users in the memory

Reply:
> I wouldn't add it to the framework since it's not used by the framework 

Disagree. Think about the app that was built on top of OC

> I think it's a bad idea to allow someone to load all these users in the memory

I mentioned above that I can return queryable users, then it's up to user to run the query or filter, or paginate the result

Reply:
YesSQL does not support IQueryable, so you can't really ask it to return IQueryable.

Reply:
I knew :) please refer to https://github.com/OrchardCMS/OrchardCore/pull/17757#discussion_r2033644379